applications/IgaApplication/custom_elements/shell_5p_hierarchic_element.h

```diff
@@ -0,0 +1,613 @@
+     */
+    void Initialize() override;
+
+    /**
+    * This functions calculates both the RHS and the LHS
+    * @param rLeftHandSideMatrix: The LHS
+    * @param rRightHandSideVector: The RHS
+    * @param rCurrentProcessInfo: The current process info instance
+    * @param CalculateStiffnessMatrixFlag: The flag to set if compute the LHS
+    * @param CalculateResidualVectorFlag: The flag to set if compute the RHS
+    */
+    void CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    );
+
+    /// @brief Sets on rResult the ID's of the element degrees of freedom
+    void EquationIdVector(
+        EquationIdVectorType& rResult,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /// returns dof list
+    void GetDofList(
+        DofsVectorType& rElementalDofList,
+        ProcessInfo& rCurrentProcessInfo
+    ) override;
+
+    /// Calls CalculateAll without computing StiffnessMatrix
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        // Calculation flags
+        const bool CalculateStiffnessMatrixFlag = false;
+        const bool CalculateResidualVectorFlag = true;
+        MatrixType temp = Matrix();
+
+        SizeType number_of_control_points = GetGeometry().size();
+        SizeType mat_size = number_of_control_points * 5;
+
+        //resizing as needed the RHS
+        if (rRightHandSideVector.size() != mat_size)
+            rRightHandSideVector.resize(mat_size);
+        rRightHandSideVector = ZeroVector(mat_size);
+
+        CalculateAll(temp, rRightHandSideVector, rCurrentProcessInfo, CalculateStiffnessMatrixFlag, CalculateResidualVectorFlag);
+    }
+
+    /// Calls CalculateAll without computing ResidualVector
+    void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+        ProcessInfo& rCurrentProcessInfo) override
+    {
+        // Calculation flags
+        const bool CalculateStiffnessMatrixFlag = true;
+        const bool CalculateResidualVectorFlag = false;
+        VectorType temp = Vector();
+
+        SizeType number_of_control_points = GetGeometry().size();
+        SizeType mat_size = number_of_control_points * 5;
+
+        //resizing as needed the LHS
+        if (rLeftHandSideMatrix.size1() != mat_size || rLeftHandSideMatrix.size2() != mat_size)
+            rLeftHandSideMatrix.resize(mat_size, mat_size);
+        noalias(rLeftHandSideMatrix) = ZeroMatrix(mat_size, mat_size);
+
+        CalculateAll(rLeftHandSideMatrix, temp, rCurrentProcessInfo, CalculateStiffnessMatrixFlag, CalculateResidualVectorFlag);
+    }
+
+    /// Calls CalculateAll
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo

```

Comment:
```suggestion
        const ProcessInfo& rCurrentProcessInfo
```

Reply:
@tteschemacher I strongly suggest to use the new interfaces, this will be deprecated & removed soon => #7381

its not much work, basically adding some const

Reply:
also I would move the implementation to the cpp

Reply:
yep. process infos are const now. const can safe lives.

Reply:
> also I would move the implementation to the cpp

I agree and disagree. To be honest this could be something which is implemented already in the base element class. It is repeated everywhere and just unnecessary lines of code...